frontend/src/pages/SignUpPage.js

```diff
@@ -0,0 +1,137 @@
+import styled from "styled-components/macro";
+import { useState } from "react";
+import useSignUp from "../hooks/useSignUp";
+import {NavLink} from "react-router-dom";
+
+export default function SignUpPage() {
+    const {checkUserName, signUp, availability, success} = useSignUp();
+    const [error, setError] = useState()
+    const [credentials, setCredentials] = useState({
+        username: "",
+        password: "",
+        passwordcheck: "",
+    });
+
+    const handleChange = (event) => {
+        setCredentials({...credentials, [event.target.name]: event.target.value});
+    };
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (credentials.password === "" || credentials.passwordcheck === "" || credentials.username === "") {
+            setError("Please fill out all fields");
+            return;
+        }
+        if (credentials.password !== credentials.passwordcheck) {
+            setError("Passwords do not match");
+            return;
+        } else {
+            checkUserName(credentials.username);
+            if (availability) {
+                setError("Username is already in use. Please choose another one");
+                return;
+            }
+            signUp(credentials.username, credentials.password)
+        }
+    };
+
+    return (
+        <div>
+            <h2>Enter your details to sign up!</h2>
+            {error && (!success) && <p>{error}</p>}
+            <LoginForm onSubmit={handleSubmit}>
+                {(!success) &&
+                    <div>
```

Comment:
Do you need this to be div? I think a react fragment is more suitable. Don't clutter up the dom with divs.